src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Remote.cs

```diff
@@ -2,18 +2,17 @@
 
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.Experiments;
 using Microsoft.CodeAnalysis.Remote;
 
 namespace Microsoft.CodeAnalysis.FindSymbols
 {
     public static partial class SymbolFinder
     {
-        internal static Task<RemoteHostClient.Session> TryGetRemoteSessionAsync(
+        internal static Task<SessionWithSolution> TryGetRemoteSessionAsync(
```

Comment:
Why does SymbolFinder need to work with one of tehse types?  Why can't it just work with a Client (as we see above in other features that remote to the OOP server)?  It can then call your helper to "RunCodeAnalysisOnRemoteXXX" (or whatever it was called).

in other words *except* for todo-comments, and NugetSearch, i would expect to see all features integrating with OOP in the same way.